src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -14,9 +14,11 @@ public partial class SocketAsyncEventArgs : EventArgs, IDisposable
         // Buffer,Offset,Count property variables.
         private WSABuffer _wsaBuffer;
         private IntPtr _ptrSingleBuffer;
+        private GCHandle _singleBufferGCHandle;
 
         // BufferList property variables.
         private WSABuffer[] _wsaBufferArray;
+        private GCHandle[] _multipleBufferGCHandles;
```

Comment:
Yeah, I thought about that too.  But since the single buffer case seems quite common, it's not clear this is worthwhile.

There's a lot of stuff we could do to make SAEA lighter weight.  I'd rather tackle some of that stuff than guess what the right tradeoff here is, since it's not obvious.